File: leetcode/python/1207.py
class Solution:
    def uniqueOccurrences(self, arr: List[int]) -> bool:
        d = defaultdict(int)
        d2 = defaultdict(bool)

        for v in arr:
            d[v] += 1

        for _, v in d.items():
            if d2[v] == True:
                return False

            d2[v] = True

        return True

File: leetcode/java/1207.java
class Solution {
  public boolean uniqueOccurrences(int[] arr) {
    Map<Integer, Integer> d = new HashMap<>();
    Map<Integer, Boolean> c = new HashMap<>();

    for (int i = 0; i < arr.length; i++) {
      if (d.containsKey(arr[i])) {
        d.put(arr[i], d.get(arr[i]) + 1);
      } else {
        d.put(arr[i], 1);
      }
    }

    for (Integer v : d.values()) {
      if (c.containsKey(v)) {
        return false;
      } else {
        c.put(v, true);
      }
    }

    return true;
  }
}
